api/v1alpha2/conversion.go

```diff
@@ -129,14 +137,40 @@ func (dst *MachineList) ConvertFrom(srcRaw conversion.Hub) error {
 
 func (src *MachineSet) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*v1alpha3.MachineSet)
+	if err := Convert_v1alpha2_MachineSet_To_v1alpha3_MachineSet(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually convert ClusterName from label, if any.
+	if name, ok := src.Labels[MachineClusterLabelName]; ok {
+		dst.Spec.ClusterName = name
+	}
+
+	// Manually restore data.
+	restored := &v1alpha3.MachineSet{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	if restored.Spec.ClusterName != "" {
+		dst.Spec.ClusterName = restored.Spec.ClusterName
```

Comment:
We set dst ClusterName above from the label if it's there. This would overwrite. Do you want to add a comment that this takes precedence?